hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/ConfigurationProvider.java

```diff
@@ -32,10 +33,17 @@
 
   private static OzoneConfiguration configuration;
 
+  private static void addDeprecations() {
+    Configuration.addDeprecation(
+        ReconServerConfigKeys.OZONE_RECON_HTTP_KEYTAB_FILE_OLD,
+        ReconServerConfigKeys.OZONE_RECON_HTTP_KEYTAB_FILE);
+  }
+
   @VisibleForTesting
   public static void setConfiguration(OzoneConfiguration conf) {
     if (configuration == null) {
       ConfigurationProvider.configuration = conf;
+      ConfigurationProvider.addDeprecations();
```

Comment:
NIT: should we move this to a static block of the class to avoid issues like caller forget to invoke setConfiguration explicitly?

  static {
    addDeprecations();  
  }

Reply:
Good suggestion! I have fixed it in the latest commit. Thanks!